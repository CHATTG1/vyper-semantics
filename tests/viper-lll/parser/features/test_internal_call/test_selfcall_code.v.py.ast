%pgm( , ,
  %fdecl(%@public, foo, , %num,
    %return(3))
  %fdecl(%@public, foo, , %num,
    %return(%icall(foo, )))
)
