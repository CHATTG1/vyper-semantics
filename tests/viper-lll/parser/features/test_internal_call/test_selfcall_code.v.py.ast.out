$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 3264763256 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , 3 ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 4273672062 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $mload ( $%seq ( $assert ( $call ( $gas , $address , 0 , $%seq ( $mstore ( 320 , 3264763256 ) , 348 ) , 4 , 416 , 32 ) ) , 416 ) ) ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
